docs/dev_contribution.rst

```diff
@@ -0,0 +1,95 @@
+
+A topic or feature branch is a branch that is local to you (and anyone you collaborate with), it is branch that will not be available from the :term:`prime repository`.  All development work should be carried out on a topic branch, for example any major feature that you work on or minor bug fix.  Before creating a local topic branch pull the latest changes from the :term:`prime repository` develop branch.
+
+.. _libcellml-branching:
+.. figure:: images/libCellMLProcesses-GitBranching.png
+   :align: center
+   :alt: Git branching illustration
+
+   Git branching illustration
+
+Following this process will make it easier to have multiple topic branches at once and keep them in-sync with the :term:`prime repository` develop branch.  Which will in turn make it easier to manage multiple pull requests.
+
+The following Git command line commands show an example of how to create a topic branch for fixing a (hypothetical) bug described in `issue #46`::
+
+   git fetch prime develop
+   git checkout develop # Not required if already on develop branch
+   git merge prime/develop
+   git checkout -b iss46
+
+Test Driven Development
+=======================
+
+Test driven development entails writing a test that covers the intended functionality (this may require a suite of tests to be written) and no more.  The tests will require some skeleton implementation so that the test(s) can compile but by definition not pass, at least not pass all the tests.  The purpose of this is two-fold: 1. writing the test(s) first sets out the intended design which can be shared through a pull request; and  2. the skeleton implementation will include the documentation clearly describing the intended purpose.  Following this contribution process allows others to comment and make corrections before time is spent on the functional code. 
+
+It may be necessary to refactor the current desgin to enable the easiest possible way to add the new feature.  This is a good thing as the quality of the design improves this makes it easier to work with in the future.  Refactoring means improving the code without adding features, and the tests provide validation that the refactored code performs as well as before.
+
+For simple or obvious bugs which have fallen through the testing gaps just the implementation is fine.
+
+GitHub Pull Request
+===================
+
+Once some changes have been made and local commits committed push your changes to your GitHub libcellml repository (refer to :numref:`Figure %s <libcellml-branching>`).  From there create a pull request from your topic branch to the :term:`prime repository` develop branch.  When creating the pull request make sure to add in the comment 'addresses issue #46' (of course replace the number 46 with the actual number of the issue you are addressing), or something to that effect.  This will create a link between the issue and the pull request enabling other people to see that you are working on this issue and comment on your work.
```

Comment:
libCellML, not libcellml (at least, that's what is done below and what I would do too).